### PR TITLE
Prevent forbidden request to `ListDeviceUnits` on guest devices

### PR DESCRIPTION
**Pull request by espenfjo**
Calling `ListDeviceUnits` on devices which you are not the owner, but are granted control will return a 403 Forbidden error.
Adding this protection should not affect normal operation in Home Assistant or similar users of this library.
For Home Assistant the only side-effect is missing model number in the unit description.

### CODE REVIEW

**Comment by vilppuvuorinen** [comment]
Looks good. I want to give it a go before merging which slows things down a touch.

Do you want to do the eventual home-assistant PR as well? This change will break the `device_info` property in the integration.

**Comment by espenfjo** [comment]
This code is "only" used for setting the model number in `device_info`:
```python
unit_infos = self.device.units
if unit_infos is not None:
    _device_info["model"] = ", ".join(
        [x["model"] for x in unit_infos if x["model"]]
    )
```
Since `self.device.units` is `None` here the code will not try to create the model attribute, and since its not already defined in the code above it will not show up as `None`, `undefined` or any other "ugly" variant.

So a PR to Homeassistant shouldn't really be necessary?

File: pymelcloud/device.py [hunk 1]
```diff
@@ -73,7 +76,7 @@ async def update(self):
         )
         self._state = await self._client.fetch_device_state(self)
 
-        if self._device_units is None:
+        if self._device_units is None and self.device_owner_id == self.user_id:
```
Comment (vilppuvuorinen) [inline pymelcloud/device.py:79]: My heat pump has `OwnerID` of `null`. I can't remember whether I've cleared the fields from the sample data, but none of the sample devices also have `null` `OwnerID`s.

**Review by vilppuvuorinen — CHANGES_REQUESTED** [review]
The CHANGLELOG should be updated. Something like

```
### Fixed
- Prevent forbidden request to `ListDeviceUnits` on guest devices.
```

You should also reword the commit message if possible, The actual message content is better than most of the commits in this repo so far, but I'd love to see the formatting follow the guidelines from this wonderful blog post: https://chris.beams.io/posts/git-commit/

**Comment by espenfjo** [comment]
I did something else instead. Rather than comparing the current user with the device owner i check the access level.

Also added some tests.

**Review by vilppuvuorinen — APPROVED** [review]

**Comment by vilppuvuorinen** [comment]
Thank you for the patch. I'll create a new release and then a PR to home assistant.

Please feel free to harass me here if either of these steps starts to drag on.

**Pull request merged**